PolyEngine/Core/Src/BTreeMap.hpp

```diff
@@ -0,0 +1,532 @@
+#pragma once
+
+#include "BTreeNode.hpp"
+#include "Optional.hpp"
+
+namespace Poly {
+
+	template<typename K, typename V, size_t B>
+	struct Range { //note(vuko): doubles as a proper fucking iterator; behold the goddamn beauty! (alright, this is for show only for now)
```

Comment:
This is a temporary code that is supposed to show off the proof-of-concept, which is why I took the liberty to use strong language, despite your stance on the idea ;)
tl;dr: wilco.